collect_app/src/androidTest/java/org/odk/collect/android/feature/projects/AddNewProjectTest.kt

```diff
@@ -3,26 +3,59 @@ package org.odk.collect.android.feature.projects
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.RuleChain
+import org.odk.collect.android.R
+import org.odk.collect.android.injection.config.AppDependencyModule
 import org.odk.collect.android.support.CollectTestRule
+import org.odk.collect.android.support.ResetStateRule
+import org.odk.collect.android.support.StubBarcodeViewDecoder
 import org.odk.collect.android.support.TestRuleChain
+import org.odk.collect.android.support.pages.MainMenuPage
+import org.odk.collect.android.views.BarcodeViewDecoder
 
 class AddNewProjectTest {
 
     val rule = CollectTestRule()
 
+    private val stubBarcodeViewDecoder = StubBarcodeViewDecoder()
+
     @get:Rule
-    var chain: RuleChain = TestRuleChain.chain().around(rule)
+    val chain: RuleChain = TestRuleChain.chain()
+        .around(
+            ResetStateRule(object : AppDependencyModule() {
+                override fun providesBarcodeViewDecoder(): BarcodeViewDecoder {
+                    return stubBarcodeViewDecoder
+                }
+            })
+        )
+        .around(rule)
 
     @Test
-    fun addingProject_addsNewProject() {
+    fun addingProjectManually_addsNewProject() {
         rule.startAtMainMenu()
             .openProjectSettings()
             .clickAddProject()
+            .switchToManualMode()
             .inputUrl("https://my-server.com")
             .inputUsername("John")
             .addProject()
 
             .openProjectSettings()
+            .assertCurrentProject("Demo project", "demo.getodk.org")
             .assertInactiveProject("my-server.com", "John / my-server.com")
     }
+
+    @Test
+    fun addingProjectAutomatically_addsNewProject() {
+        val page = rule.startAtMainMenu()
+            .openProjectSettings()
+            .clickAddProject()
+
+        stubBarcodeViewDecoder.scan("{\"general\":{\"server_url\":\"https:\\/\\/my-server.com\",\"username\":\"adam\",\"password\":\"1234\"},\"admin\":{}}")
+        page.checkIsToastWithMessageDisplayed(R.string.new_project_created)
```

Comment:
I think the "project successfully added" feedback is nice. Since we're not switching to the project that was just created, it is unclear whether anything happened. Probably what we'd want to work towards is a snackbar with a "Switch" action or something like that but I think for now we can go with this toast.

I'm less convinced by the project switching toast. I think once the app bar has the project title (#4584) it will feel redundant. Let's leave it for now and see how it feels with #4584.